src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -49,14 +49,15 @@ const logStateTransition = (
   tookMs: number
 ) => {
   if (newState.logs.length > oldState.logs.length) {
+    const doLog = (level: MigrationLogLevel, message: string) => {
+      if (level === 'error') {
+        logger.error(message);
+      } else {
```

Comment:
optional nit: let's add an exhaustive check to make sure we don't forget to update this code when add support for a new logging level in the future:
```js
      if (log.level === 'error') {
        logger.error(message);
      } else if (log.level === 'info') {
        logger.info(message);
      } else {
        throw new Error(`unexpected log level ${log.level}`);
      }
```